components/Modals/Window/window.js

```diff
@@ -0,0 +1,53 @@
+import { useRef, useEffect } from "react";
+import ReactDOM from "react-dom";
+import styled from "styled-components";
+
+export default function Window({ children, onClose }) {
+  const modalWindow = useRef();
+
+  function backDropHandler(event) {
+    if (!modalWindow.current.contains(event.target)) {
+      onClose();
+    }
+  }
+
+  useEffect(() => {
+    setTimeout(() => {
+      window.addEventListener("click", backDropHandler);
+    });
```

Comment:
I think you don't need the setTimeout here. You should be able to add the eventlistener directly after the model is rendered.

Reply:
I tried it. It really does not work without the timeout!